crates/rpc-client/src/rpc_client.rs

```diff
@@ -728,6 +730,277 @@ impl RPCClient {
+            withdrawal_capacity.saturating_add(custodian_change_capacity)
+        };
+
+        while candidate_capacity < required_capacity
+            || candidate_fulfilled_sudts < withdrawals_amount.sudt.len()
+            || candidate_cells < max_custodian_cells
+        {
+            let cells: Pagination<Cell> = to_result(
+                self.indexer
+                    .client()
+                    .request(
+                        "get_cells",
+                        Some(ClientParams::Array(vec![
+                            json!(search_key),
+                            json!(order),
+                            json!(limit),
+                            json!(cursor),
+                        ])),
+                    )
+                    .await?,
+            )?;
+
+            if cells.last_cursor.is_empty() {
+                break;
+            }
+            cursor = Some(cells.last_cursor);
+
+            for cell in cells.objects.into_iter() {
+                let args = cell.output.lock.args.clone().into_bytes();
+                let custodian_lock_args = match CustodianLockArgsReader::verify(&args[32..], false)
+                {
+                    Ok(()) => CustodianLockArgs::new_unchecked(args.slice(32..)),
+                    Err(_) => continue,
+                };
+
+                if custodian_lock_args.deposit_block_number().unpack() > last_finalized_block_number
+                {
+                    continue;
+                }
+
+                let opt_sudt_type_script = cell.output.type_.clone().map(|json_script| {
+                    let script = ckb_types::packed::Script::from(json_script);
+                    Script::new_unchecked(script.as_bytes())
+                });
+
+                // Verify sudt
+                if let Some(sudt_type_script) = opt_sudt_type_script.as_ref() {
+                    // Invalid custodian type script
+                    let l1_sudt_script_type_hash =
+                        rollup_context.rollup_config.l1_sudt_script_type_hash();
+                    if sudt_type_script.code_hash() != l1_sudt_script_type_hash
+                        || sudt_type_script.hash_type() != ScriptHashType::Type.into()
+                    {
+                        continue;
+                    }
+
+                    if sudt_type_script.hash() != CKB_SUDT_SCRIPT_ARGS
+                        && parse_sudt_amount(&cell).is_err()
+                    {
+                        log::error!("invalid sudt amount, out_point: {:?}", cell.out_point);
+                        continue;
+                    }
+                }
+
+                let (sudt_amount, type_hash) = match opt_sudt_type_script.as_ref() {
+                    Some(h) if h.hash() != CKB_SUDT_SCRIPT_ARGS => {
+                        (parse_sudt_amount(&cell).expect("sudt amount"), h.hash())
+                    }
+                    _ => (0, CKB_SUDT_SCRIPT_ARGS),
```

Comment:
IMO we should simply skip these custodian cells which hash equals to CKB_SUDT_SCRIPT_ARGS.

Reply:
Fixed.